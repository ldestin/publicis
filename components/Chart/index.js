import React, { useRef, useEffect, useState } from 'react'
import { useMemo } from 'react'
import PropTypes from 'prop-types'
import chart from 'chart.js'
import Hit from '../Hit'

const Chart = ({ hits }) => {
  const ref = useRef()
  const data = useMemo(() => hits.map(hit => hit.points || 0), [hits])
  const labels = useMemo(() => hits.map(hit => hit.objectID), [hits])

  useEffect(() => {
    new chart(ref.current, {
      type: 'line',
      options: {
        scales: {
          xAxes: [{
            scaleLabel: {
              display: true,
              labelString: 'ID',
            }
          }],
          yAxes: [{
            scaleLabel: {
              display: true,
              labelString: 'Votes',
            }
          }]
        },
        responsive: true,
        animation: {
          duration: 0
        },
      },
      data: {
        datasets: [
          {
            label: 'Current hits',
            data,
            borderColor: '#0D47A1',
            backgroundColor: 'transparent',
            lineTension: 0,
          }
        ],
        labels,
        fill: false,
      }
    })
  }, [ref.current, hits])

  return (
    <div className="chart">
      <canvas ref={ref} width={400} height={100}></canvas>
    </div>
  )
}

Chart.defaultProps = {
  hits: [],
}

Chart.propTypes = {
  hits: PropTypes.arrayOf(Hit.propTypes.hit),
}

export default Chart
