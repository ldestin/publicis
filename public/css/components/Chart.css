.chart {
  padding: 1rem 2rem;
}

.chart canvas {
  background: var(--white)
}

@media (max-width: 1000px) {
  .chart {
    display: none;
  }
}
