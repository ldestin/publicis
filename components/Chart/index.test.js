import React from 'react'
import chart from 'chart.js'
import { render } from '@testing-library/react'
import Chart from '.'

jest.mock('chart.js')

const hits = [
  { objectID: 'id 1', points: 123 },
  { objectID: 'id 2', points: 45 },
]

const expectedConfig = {
  data: {
    datasets: [
      {
        backgroundColor: 'transparent',
        borderColor: '#0D47A1',
        data: [123, 45],
        label: 'Current hits',
        lineTension: 0
      }
    ],
    fill: false,
    labels: ['id 1', 'id 2']
  },
  options: {
    animation: {
      duration: 0
    }, 
    responsive: true,
    scales: {
      xAxes: [
        {
          scaleLabel: {
            display: true,
            labelString: 'ID'
          }
        }
      ],
      yAxes: [
        {
          scaleLabel: {
            display: true,
            labelString: 'Votes'
          }
        }
      ]
    }
  },
  type: 'line'
}

it('should initialize chart correctly', () => {
  const { container } = render(<Chart hits={hits} />)
  expect(chart).toHaveBeenCalledWith(expect.any(HTMLCanvasElement), expectedConfig)
  expect(container).toMatchSnapshot()
})
