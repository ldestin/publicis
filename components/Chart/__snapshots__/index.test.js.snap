// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should initialize chart correctly 1`] = `
<div>
  <div
    class="chart"
  >
    <canvas
      height="100"
      width="400"
    />
  </div>
</div>
`;
